docs/src/manuals/meta-specification.md

```diff
@@ -57,35 +63,185 @@ List of available options:
+
+```julia
+inferred_result = inference(
+    model = my_model(arguments...),
+    meta = my_meta,
+    ...
+)
+
+inferred_result = rxinference(
+    model = my_model(arguments...),
+    meta = my_meta,
+    ...
+)
+```
+
+**Note**: You can also specify metadata for your nodes directly inside `@model`, without the need to use `@meta`. For example:
+```julia
+@model function my_model()
+    ...
+
+    y ~ AR(x, θ, γ) where { meta = ARMeta(Univariate, 5, ARsafe()) }
+
+    ...
+end
+```
+If you add node-specific meta to your model this way, then you do not need to use the `meta` keyword argument in the `inference` and `rxinference` functions.
+
+
+## Create your own meta
+Although some nodes in `RxInfer.jl` already come with their own meta structure, you have the flexibility to define different meta structures for those nodes and also for your custorm ones. A meta structure is created by using the `struct` statement in `Julia`. For example, the following snippet of code illustrates how you can create your own meta structures for your custom node:
+
+```julia
+# create your own meta structure for your custom node
+struct MyCustomMeta
+    arg1 
+    arg2 
+end
+
+# apply the new meta structure to your node
+@meta function model_meta(arg1, arg2)
+    MyCustomNode() -> MyCustomMeta(arg1,arg2)
+end
+
+my_meta = model_meta(value_arg1, value_arg2)
+```
+or create different meta structures for a node, e.g. `AR` node:
+```julia
+# create your own meta structure for the AR node
+struct MyARMeta
+    arg1
+    arg2
+end
+
+# apply the new meta structure to the AR node
+@meta function model_meta(arg1, arg2)
+    AR() -> MyARMeta(arg1, arg2)
+end
+
+my_meta = model_meta(value_arg1, value_arg2)
+```
+**Note**: When you define a meta structure for a node, keep in mind that you must define message update rules with the meta for that node. See [node](@id create-node) for more details of how to define rules for a node.
+
+## Example
+This section provides a concrete example of how to create and use meta in `RxInfer.jl`. Suppose that we have the following Gaussian model:
+
+$$\begin{aligned}
+ x & \sim \mathrm{Normal}(2.5, 0.5)\\
+ y & \sim \mathrm{Normal}(2*x, 2.0)
+\end{aligned}$$
+
+where $y$ is observable data and $x$ is a latent variable. In `RxInfer.jl`, the inference procedure for this model is well defined without the need of specifying any meta data for the `Normal` node.
+```julia
+using RxInfer
+
+#create data
+y_data = 4.0 
+
+#make model
+@model function gaussian_model()
+    y = datavar(Float64)
+
+    x ~ NormalMeanVariance(2.5, 0.5)
+    y ~ NormalMeanVariance(2*x, 2.)
+end
+
+#do inference
+inference_result = inference(
+    model = gaussian_model(),
+    data = (y = y_data,)
+)
+```
+However, let's say we would like to define a new inference procedure by introducing a meta structure to the `Normal` node that always yields a Normal distribution with mean $m$ for the outbound messages of the node. This is done as follows:
+```julia
```

Comment:
Can we use 
```suggestion
```@example custom-meta
```
instead? The `@example` instead of `julia` will actually run the code and will throw an error if it fails.

Reply:
See hre https://documenter.juliadocs.org/stable/man/syntax/#@example-block